### PR TITLE
Track to wait for order to be created

### PR DESCRIPTION
**Pull request by JonnyCarter**

### CODE REVIEW

**Comment by JonnyCarter** [comment]
//Find the fastest way of checking if an order exists
        //Do for max 5 seconds if 5 seconds pass move on
        //As soon as found move on 
        //For slow servers this code keeps us waiting before submitting the success page

File: app/code/community/Divido/Pay/controllers/PaymentController.php [hunk 1]
```diff
@@ -286,7 +286,19 @@ public function returnAction ()
 
         $session->setLastQuoteId($quoteId)->setLastSuccessQuoteId($quoteId);
 
-        $order = Mage::getModel('sales/order')->loadByAttribute('quote_id', $quoteId);
+        while($i < 5) {
```
Comment (cjgk) [inline app/code/community/Divido/Pay/controllers/PaymentController.php]: Could $i not being declared lead to notices?

File: app/code/community/Divido/Pay/controllers/PaymentController.php [hunk 2]
```diff
@@ -286,7 +286,19 @@ public function returnAction ()
...
+            $order = Mage::getModel('sales/order')->loadByAttribute('quote_id', $quoteId);
+            if ($order->getId()) {
+                Mage::log('already have order with id ' . $quoteId, Zend_Log::DEBUG, 'divido.log', true);
+                $i=6;
```
Comment (cjgk) [inline app/code/community/Divido/Pay/controllers/PaymentController.php]: Leave the loop with break; so editing the total waiting time don't require us to remember changing this row too.

File: app/code/community/Divido/Pay/controllers/PaymentController.php (hunk 1 above)
Comment (cjgk) [inline app/code/community/Divido/Pay/controllers/PaymentController.php]: Could we put `5` in a class constant?

File: app/code/community/Divido/Pay/controllers/PaymentController.php [hunk 3]
```diff
@@ -286,7 +286,19 @@ public function returnAction ()
 
         $session->setLastQuoteId($quoteId)->setLastSuccessQuoteId($quoteId);
 
-        $order = Mage::getModel('sales/order')->loadByAttribute('quote_id', $quoteId);
+        while($i < 5) {
+            $order = Mage::getModel('sales/order')->loadByAttribute('quote_id', $quoteId);
+            if ($order->getId()) {
+                Mage::log('already have order with id ' . $quoteId, Zend_Log::DEBUG, 'divido.log', true);
```
Comment (cjgk) [inline app/code/community/Divido/Pay/controllers/PaymentController.php]: Check if we have the debug flag on in the plugin before logging.

File: app/code/community/Divido/Pay/controllers/PaymentController.php [hunk 4]
```diff
@@ -286,7 +286,19 @@ public function returnAction ()
...
+                $i=6;
+            }else { 
+                Mage::log('order not created waiting: ' . $quoteId, Zend_Log::DEBUG, 'divido.log', true);
```
Comment (cjgk) [inline app/code/community/Divido/Pay/controllers/PaymentController.php]: Check if we have the debug flag on in the plugin before logging.

**Comment by JonnyCarter** [comment]
Suggestions have been addressed

**Pull request merged**